File: problems/2645-pass-the-pillow/pass-the-pillow.py
class Solution:
    def passThePillow(self, n: int, time: int) -> int:
        i = 0
        currentTime = 0
        up = True
        while currentTime <= time:
            if up == True:
                i += 1
            else:
                i -= 1

            if up == True and i == n:
                up = False
            if up == False and i == 1:
                up = True
                
            currentTime += 1
            
        return i

File: problems/2645-pass-the-pillow/pass-the-pillow.js
/**
 * @param {number} n
 * @param {number} time
 * @return {number}
 */
const passThePillow = (n, time) => {
    let i = 0;
    let current = 0;
    let directionUp = true;
    while (current <= time) {
        if (directionUp && i === n) {
            directionUp = false;
        }
        if (!directionUp && i === 1) {
            directionUp = true;
        }
        directionUp ? i++ : i--;
        current ++;
    }
    return i;
};